### DISCUSSION TITLE
Draw a circle of icons to represent Peer-to-Peer cluster technologies like Elasticsearch and Cassandra?

### DISCUSSION THREAD

**Discussion by HariSekhon** [terrastruct/d2#1330] [General]
Hi,

I'd like to make diagrams containing clusters of Elasticsearch and Cassandra which have peer-to-peer ring architecture, but I can't see how to generate this in D2.

Is this something that is currently possible or should I raise a Feature Request for it?

**Comment by alixander** [comment]
that's interesting. i think this is common enough in software arch diagrams that we should do this. 

we can continue down the line of special sub-diagrams (like grid and sequence), and introduce a `shape: radial`. wdyt @gavin-ts ?

**Reply by gavin-ts** [reply]
I think that makes sense, maybe a hub and spoke diagram type to make diagrams like the following:

![](https://as1.ftcdn.net/v2/jpg/05/88/60/56/1000_F_588605600_D6WS6IjOBLK8CQ3JOK8nPApwHjWSXOsW.jpg)

**Reply by alixander** [reply]
i feel like this diagram is unconventional, but in general hub-and-spoke where N connected 1 center thing makes sense.

**Comment by HariSekhon** [comment]
Only the selected nodes should be in a circle eg. for Cassandra or Elasticsearch or similar peer clusters - the rest of the nodes would need to behave normally so that these clusters could be integrated into real-world representative looking diagrams.
